test/test_autograd.py

```diff
@@ -719,6 +719,30 @@ def backward(ctx, x):
         torch.autograd.backward(x, inputs=(y, ))  # allow_unused is implicitly True!
         self.assertIsNone(y.grad)
 
+    def test_grad_batched_grad(self):
+        x = torch.randn(2, 2, requires_grad=True)
+
+        out = x.clone()  # Size([2, 2])
+        batched_grad = torch.arange(3).expand(2, 2, 3).transpose(0, 2)  # Size([3, 2, 2])
+        grad, = torch.autograd.grad(out, (x,), (batched_grad,), is_grads_batched=True)
+        self.assertEqual(grad, torch.arange(3).expand(2, 2, 3).transpose(0, 2).to(dtype=grad.dtype))
+
+        # Detect shape mismatch
+        grad_out = torch.ones(2, 2)
+        with self.assertRaisesRegex(RuntimeError, "If `is_grads_batched=True`, we interpret the first"):
+            torch.autograd.grad(outputs=out, grad_outputs=(grad_out,), inputs=(x,), is_grads_batched=True)
```

Comment:
If vmap ends up in pytorch core, would we still want to have the `is_grads_batched=True` API? Because you can do this with by using vmap:
```
vmap(torch.autograd.grad, (None, None, 0))(out, x, grad_out)
```

Reply:
One reason for having both is that some users might be more comfortable with imperative APIs over a functional one,
though I can see that its not super ideal to have two ways of doing the same thing.

Reply:
Paging @albanD for discussion. Either way I'm fine with this going in as-is (we can always revisit the messaging later...)

Reply:
I think this is a good question. I think it is ok to have both, especially because this one is not necessarily related to vmap. And I can see people using this one without even knowing what vmap is.